### PR TITLE
Fix Unit tests to be compatible with php8

### PR DESCRIPTION
**Pull request by anzin**
### Description (*)

Fixes existing Unit Tests (and related files from Magento Core) to be compatible both with PHP7 and PHP8.

### Related Issues
1. magento/magento2#34019

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by anzin** [comment]
@magento run Unit Tests

**Comment by anzin** [comment]
@magento run Unit Tests with env PHP 8.0

**Review by ihor-sviziev — APPROVED** [review]

**Review by eduard13 — APPROVED** [review]

**Review by andrewbess — APPROVED** [review]

File: app/code/Magento/Ups/Test/Unit/Model/CarrierTest.php [hunk 1]
```diff
@@ -445,8 +445,25 @@ public function testRequestToShipment(array $requestData, array $expectedRequest
...
+            ->method('getOrder')
+            ->willReturn($orderMock);
+        $orderMock->expects($this->any())
+            ->method('getIncrementId')
+            ->willReturn('100000001');
+
+        $requestData['order_shipment'] = $shipmentMock;
         $request->setData($requestData);
-        $request->setPackages([['items' => []]]);
+        $request->setPackages([['items' => [], 'params' => ['container' => '']]]);
```
Comment (xmav) [inline app/code/Magento/Ups/Test/Unit/Model/CarrierTest.php:466]: Could you please check on a live Magento that those array keys always present and we will not hit same issue in runtime?
https://github.com/magento/magento2/blob/1807f70756b521240c06eb556802422307cabbb5/app/code/Magento/Ups/Model/Carrier.php#L1523-L1524

**Review by xmav — CHANGES_REQUESTED** [review]

**Comment by andrewbess** [comment]
@magento give me test instance with env PHP 8.0

File: app/code/Magento/Ups/Test/Unit/Model/CarrierTest.php (hunk 1 above)
Comment (anzin) [reply app/code/Magento/Ups/Test/Unit/Model/CarrierTest.php:466]: Hello, @xmav  thanks for review, I checked on a live Magento that those array keys always presents, they are added to the request from the configuration, that's why we will not have this problem in runtime.

Thank you.

**Pull request merged**